internal/invocation/invocation_service.go

```diff
@@ -183,7 +187,16 @@ func (s *Service) handleClientMessage(msg *proto.ClientMessage) {
 				return fmt.Sprintf("invocation with unknown correlation ID: %d", correlationID)
 			})
 		} else if inv.EventHandler() != nil {
-			go inv.EventHandler()(msg)
+			handler := func() {
+				inv.EventHandler()(msg)
+			}
+			if inv.PartitionID() == -1 {
```

Comment:
How is it possible? I couldn't find such case if I don't miss something.

Reply:
When an invocation does not have a specific target partition, partitionID is set to "-1". I have read it on Java client, and see it for my self on manual tests, but sadly I cannot point you to an actual doc.

Please check reference [Java implementation](https://github.com/hazelcast/hazelcast/blob/fb720d9e7f4b7643c31fdb61e9b83b6ca8ecc552/hazelcast/src/main/java/com/hazelcast/internal/util/executor/StripedExecutor.java)

Reply:
Refactored -1 to a constant for readability 👍